### DISCUSSION TITLE
[General] eth recharge confirmation (deposit finality)

### DISCUSSION THREAD

**Discussion by Poetry-Distance** [zkSync-Community-Hub/zksync-developers#697] [General] [answered]
### Environment

Mainnet

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

According to the previous conclusion, ETH recharge needs to be confirmed 10 times on the L1 (ETH) network before it will be rolled back.

The current issue is whether the RPC fields returned by zksync indicate how many times such fields have been confirmed on L1, or whether there are fields indicating that this transaction will not be rolled back?

If not, what is the effective logic for the maturity confirmation of zksync transactions that you recommend? As a centralized exchange, we want to confirm the legality of users' recharge transactions based on this logic.

**Comment by alhenaai** [comment]
In ZKsync, the confirmation of transactions on the L1 (Ethereum) network is crucial for ensuring the finality and security of transactions. However, the RPC fields returned by ZKsync do not directly indicate the number of confirmations on L1 for a specific transaction. Instead, transactions in ZKsync follow a lifecycle where they are first pending, then included, and finally verified or failed.

For a centralized exchange, the recommended approach to confirm the legality and finality of users' recharge transactions is to track the transaction status through the ZKsync RPC interface. The key statuses to monitor are:
- **Included**: The transaction is included in an L2 block.
- **Verified**: The transaction has been batched, proven, and the proof has been accepted on L1, ensuring it cannot be rolled back.

You should consider a transaction as final and irreversible once it reaches the "Verified" status. This status means that the transaction has been batched into a proof, and that proof has been successfully posted and accepted on the Ethereum mainnet.

For more detailed information on transaction statuses and the transaction lifecycle in ZKsync, you can refer to the [Transaction Lifecycle documentation](https://docs.zksync.io/zk-stack/concepts/transaction-lifecycle#transaction-statuses).

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by uF4No** [comment, marked as answer]
I for deposits, it'd the same as on Ethereum (2 epochs or ~ 15mins) as user just transfers tokens to the ZKsync contract in L1.

For withdraws, given that the system still has a 24h withdraw delay enabled it'd be around that.

Please check: 
- [Finality on ZKsync](https://docs.zksync.io/zk-stack/concepts/finality)
- [Withdraw delay](https://docs.zksync.io/build/resources/withdrawal-delay)
